index.js

```diff
@@ -59,6 +59,7 @@ const hasProperties = curry((propertyTypes, object) => {
         type,
         required = false
       } = propertyType
+      if (!required) return true
```

Comment:
this changes everything that's not required into an `any` type, this means you can put anything in a non-required field and it's valid. instead i think we should check if it's not required and you put a nil value, it's good.

Reply:
oh good thought @ahdinosaur - of course - will look at this again 🎉